### ISSUE TITLE
SSD model: Get very low result on VOC 2007 test data, with enough training steps

### ISSUE THREAD

**Issue by jfojfo** [tryolabs/luminoth#166]
does SSD model work now ? 
got the following result with default ssd config
Average Precision (AP) @ [0.50] = 0.523

trained with enough steps

**Comment by joaqo** [comment]
hey @jfojfo ! Would you mind telling me how many steps you run it for? And at what learning rates?

On my latest train I got 0.70 on our @.50 metric and 0.755 on pascal's official metric, which filters out hard examples and is what they used on the paper to get their 0.772 mAP. Keep in mind that this version is hardcoded to batch_size = 1, so you'll need many more steps than what they quote on the paper. I am currently implementing a PR which allows you to choose your batch size, so that'll be fixed soon.

Also, choosing a non optimal batch size has been shown to decrease final performance, so I hope allowing us to choose batch size will let us get that extra 0.017 we are missing to get to the paper's performance!

**Comment by jfojfo** [comment]
@joaqo thanks for reply.  I commented out 'expand' and train ssd for 709389 steps and got following result:
INFO:tensorflow:Finished evaluation at step 709389.
INFO:tensorflow:Evaluated 4952 images.
INFO:tensorflow:Average Precision (AP) @ [0.50] = 0.586
INFO:tensorflow:Average Precision (AP) @ [0.75] = 0.280
INFO:tensorflow:Average Precision (AP) @ [0.50:0.95] = 0.302
INFO:tensorflow:Average Recall (AR) @ [0.50:0.95] = 0.477

should I continue to train with more steps?

**Comment by joaqo** [comment]
@jfojfo I trained with this schedule:

- 1.700.000 steps @ 0.0003 learning rate
- 1.700.000 - 2.000.000 @ 0.0001 learning rate
- 2.000.000 - 3.000.000 @ 0.00001 learning rate

I realize this is probably overkill, but I didn't have an eval running so I couldn't see how the validation error progressed, and I was using batch size 1 instead of 32 as in the paper, so I just left the machine training through the weekend to be sure. You can probably use much fewer steps and get the same results. 

One thing i noticed is that you didn't lower your learning rate, this is crucial, so the first thing I would try if I were you would be to take that model you trained, and lower the learning rate from `0.0003` to `0.0001` and continue training for a couple hundred thousand steps.
